tests/src/Simple/Generics/Generics.cs

```diff
@@ -567,4 +568,221 @@ public static void Run()
+                s_IFooInt = "GenBase<System.Byte>.IMethod1<System.Int32>(7,8)";
+                TestWithGenClass<byte>(new GenBase<byte>());
+                Console.WriteLine("====================");
+
+
+                s_GMethod1 = "GenDerived<System.Byte>.GMethod1<System.Int32>(1,2)";
+                s_IFooString = "GenDerived<System.Byte>.IFoo<string>.IMethod1<System.Int32>(3,4)";
+                s_IFooObject = "GenDerived<System.Byte>.IFoo<string>.IMethod1<System.Int32>(5,6)";
+                s_IFooInt = "GenBase<System.Byte>.IMethod1<System.Int32>(7,8)";
+                TestWithGenClass<byte>(new GenDerived<byte>());
+                Console.WriteLine("====================");
+
+
+                s_GMethod1 = "GenDerived<System.String>.GMethod1<System.Int32>(1,2)";
+                s_IFooString = "GenDerived<System.String>.IFoo<string>.IMethod1<System.Int32>(3,4)";
+                s_IFooObject = "GenDerived<System.String>.IFoo<string>.IMethod1<System.Int32>(5,6)";
+                s_IFooInt = "GenBase<System.String>.IMethod1<System.Int32>(7,8)";
+                TestWithGenClass<String>(new GenDerived<String>());
+                Console.WriteLine("====================");
+
+
+                s_GMethod1 = "GenDerived<System.Byte>.GMethod1<System.Int32>(1,2)";
+                s_IFooString = "GenDerived<System.Byte>.IFoo<string>.IMethod1<System.Int32>(3,4)";
+                s_IFooObject = "GenDerived<System.Byte>.IFoo<string>.IMethod1<System.Int32>(5,6)";
+                s_IFooInt = "GenSuperDerived<System.Byte>.IFoo<int>.IMethod1<System.Int32>(7,8)";
+                TestWithGenClass<byte>(new GenSuperDerived<byte>());
+                Console.WriteLine("====================");
+            }
+
+            {
+                s_IFooString = "MyStruct1.IFoo<string>.IMethod1<System.Int32>(1,2)";
+                s_IFooObject = "MyStruct1.IFoo<string>.IMethod1<System.Int32>(3,4)";
+                s_IFooInt = "MyStruct1.IFoo<int>.IMethod1<System.Int32>(5,6)";
+                TestWithStruct(new MyStruct1(), new MyStruct1(), new MyStruct1());
+                Console.WriteLine("====================");
+
+
+                s_IFooString = "MyStruct2.IFoo<string>.IMethod1<System.Int32>(1,2)";
+                s_IFooObject = "MyStruct2.IFoo<string>.IMethod1<System.Int32>(3,4)";
+                s_IFooInt = "MyStruct2.IMethod1<System.Int32>(5,6)";
+                TestWithStruct(new MyStruct2(), new MyStruct2(), new MyStruct2());
+                Console.WriteLine("====================");
+
+
+                s_IFooString = "MyStruct3.IMethod1<System.Int32>(1,2)";
+                s_IFooObject = "MyStruct3.IMethod1<System.Int32>(3,4)";
+                s_IFooInt = "MyStruct3.IFoo<int>.IMethod1<System.Int32>(5,6)";
+                TestWithStruct(new MyStruct3(), new MyStruct3(), new MyStruct3());
+                Console.WriteLine("====================");
+            }
+
+            if (s_NumErrors == 0)
```

Comment:
Yes :-) The test didn't reach this point yet. I hit a fail fast much earlier than this.